common/component/redis/redis.go

```diff
@@ -252,3 +255,65 @@ type RedisError string
 func (e RedisError) Error() string { return string(e) }
 
 func (RedisError) RedisError() {}
+
+func (s *Settings) refreshTokenRoutineForRedis(ctx context.Context, redisClient RedisClient, version string, meta map[string]string, logger logger.Logger) {
+	ticker := time.NewTicker(tokenRefreshInterval)
+	defer ticker.Stop()
+
+	if !s.useAzureAD {
```

Comment:
I would like to see this exit condition instead be used in the v8client and v9client only.

```go
if s.useAzureAd {
  authenticateWithAzureRedis()
}
```

or something like that